src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.MemoryTests;
+using Xunit;
+
+namespace System.Memory.Tests
+{
+    public class CommonReadOnlySequenceTests
+    {
+        [Fact]
+        public void SegmentStartIsConsideredInBoundsCheck()
+        {
+            // 0               50           100    0             50             100
+            // [                ##############] -> [##############                ]
+            //                         ^c1            ^c2
+            var bufferSegment1 = new BufferSegment(new byte[49]);
+            BufferSegment bufferSegment2 = bufferSegment1.Append(new byte[50]);
+
+            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment2, 50);
+
+            SequencePosition c1 = buffer.GetPosition(buffer.Start, 25); // segment 1 index 75
+            SequencePosition c2 = buffer.GetPosition(buffer.Start, 55); // segment 2 index 5
+
+            ReadOnlySequence<byte> sliced = buffer.Slice(c1, c2);
+            Assert.Equal(30, sliced.Length);
+        }
+
+        [Fact]
+        public void GetPositionPrefersNextSegment()
+        {
+            BufferSegment bufferSegment1 = new BufferSegment(new byte[50]);
+            BufferSegment bufferSegment2 = bufferSegment1.Append(new byte[0]);
+
+            ReadOnlySequence<byte> buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment2, 0);
+
+            SequencePosition c1 = buffer.GetPosition(buffer.Start, 50);
+
+            Assert.Equal(0, c1.Index);
+            Assert.Equal(bufferSegment2, c1.Segment);
+        }
+
+        [Fact]
+        public void GetPositionDoesNotCrossOutsideBuffer()
+        {
+            var bufferSegment1 = new BufferSegment(new byte[100]);
+            BufferSegment bufferSegment2 = bufferSegment1.Append(new byte[100]);
+            BufferSegment bufferSegment3 = bufferSegment2.Append(new byte[0]);
+
+            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment2, 100);
+
+            SequencePosition c1 = buffer.GetPosition(buffer.Start, 200);
+
+            Assert.Equal(100, c1.Index);
+            Assert.Equal(bufferSegment2, c1.Segment);
+        }
+
+        [Fact]
+        public void Create_WorksWithArray()
+        {
+            var buffer = new ReadOnlySequence<byte>(new byte[] { 1, 2, 3, 4, 5 });
+            Assert.Equal(buffer.ToArray(), new byte[] {  1, 2, 3, 4, 5 });
```

Comment:
nit: extra white space (and here - https://github.com/dotnet/corefx/pull/26694/files#diff-d174315954f1013bbba56cde4fd50e17R177)
```C#
Assert.Equal(buffer.ToArray(), new byte[] {  1, 2, 3, 4, 5 });
```